lib/std/os/windows.zig

```diff
@@ -135,6 +135,7 @@ pub fn OpenFileW(
         .SecurityQualityOfService = null,
     };
     var io: IO_STATUS_BLOCK = undefined;
+    const create_options = if (std.io.is_async) FILE_NON_DIRECTORY_FILE else FILE_NON_DIRECTORY_FILE | FILE_SYNCHRONOUS_IO_NONALERT;
```

Comment:
This should actually be a parameter to OpenFileW I think. Have a look at `std.fs.Dir.openFileW` and the corresponding posix function `openFileZ`, noting the use of `async_block_allowed` field.